qiime2_pipeline.bash

```diff
@@ -0,0 +1,169 @@
+cp $IDX $EMP_PAIRED_END_SEQUENCES_DIR/"barcodes.fastq.gz"
+
+qiime tools import \
+  --type EMPPairedEndSequences \
+  --input-path $EMP_PAIRED_END_SEQUENCES_DIR \
+  --output-path $PROJECT_DIR"/emp-paired-end-sequences.qza"
+
+###=====================
+### DEMULTIPLEXING SEQUENCE
+###=====================
+
+qiime demux emp-paired \
+  --m-barcodes-file $MAPPING_FP \
+  --m-barcodes-category BarcodeSequence \
+  --i-seqs $PROJECT_DIR"/emp-paired-end-sequences.qza" \
+  --o-per-sample-sequences $PROJECT_DIR"/demux" \
+  --p-rev-comp-mapping-barcodes
+
+qiime demux summarize \
+  --i-data $PROJECT_DIR"/demux.qza" \
+  --o-visualization $PROJECT_DIR"/demux.qzv"
+
+qiime tools export \
+  $PROJECT_DIR"/demux.qzv" \
+  --output-dir $PROJECT_DIR"/demux"
+
+###=====================
+###  SEQUENCE QC AND FEATURE TABLE
+###=====================
+
+qiime dada2 denoise-paired \
+  --i-demultiplexed-seqs $PROJECT_DIR"/demux.qza" \
+  --p-trim-left-f 0 \
+  --p-trunc-len-f 230 \
+  --p-trim-left-r 0 \
+  --p-trunc-len-r 230 \
+  --p-n-threads 8 \
+  --o-representative-sequences $PROJECT_DIR"/rep-seqs.qza" \
+  --o-table $PROJECT_DIR"/table.qza"
+
+qiime feature-table summarize \
+  --i-table $PROJECT_DIR"/table.qza" \
+  --o-visualization $PROJECT_DIR"/table.qzv" \
+  --m-sample-metadata-file $MAPPING_FP
+
+qiime feature-table tabulate-seqs \
+  --i-data $PROJECT_DIR"/rep-seqs.qza" \
+  --o-visualization $PROJECT_DIR"/rep-seqs.qzv"
+
+qiime tools export \
+  $PROJECT_DIR"/table.qzv" \
+  --output-dir $PROJECT_DIR"/table"
+
+qiime tools export \
+  $PROJECT_DIR"/table.qza" \
+  --output-dir $PROJECT_DIR"/table"
+
+###=====================
+###  TAXONOMIC ANALYSIS
+###=====================
+
+qiime feature-classifier classify-sklearn \
+  --i-classifier $gg_FP \
+  --i-reads $PROJECT_DIR"/rep-seqs.qza" \
+  --o-classification $PROJECT_DIR"/taxonomy.qza"
+
+qiime metadata tabulate \
+  --m-input-file $PROJECT_DIR"/taxonomy.qza" \
+  --o-visualization $PROJECT_DIR"/taxonomy.qzv"
+
+qiime tools export \
+  $PROJECT_DIR"/taxonomy.qza" \
+  --output-dir $PROJECT_DIR"/taxonomy"
+
+###=====================
+###  GENERATE TREES
+###=====================
+
+qiime alignment mafft \
+  --i-sequences $PROJECT_DIR"/rep-seqs.qza" \
+  --o-alignment $PROJECT_DIR"/aligned-rep-seqs.qza"
+
+qiime alignment mask \
+  --i-alignment $PROJECT_DIR"/aligned-rep-seqs.qza" \
+  --o-masked-alignment $PROJECT_DIR"/masked-aligned-rep-seqs.qza"
+
+qiime phylogeny fasttree \
+  --i-alignment $PROJECT_DIR"/masked-aligned-rep-seqs.qza" \
+  --o-tree $PROJECT_DIR"/unrooted-tree.qza"
+
+qiime phylogeny midpoint-root \
+  --i-tree $PROJECT_DIR"/unrooted-tree.qza" \
+  --o-rooted-tree $PROJECT_DIR"/rooted-tree.qza"
+
+###=====================
+###  ALPHA AND BETA DIVERSITY
+###=====================
+
+qiime diversity core-metrics \
+  --i-phylogeny $PROJECT_DIR"/rooted-tree.qza" \
+  --i-table $PROJECT_DIR"/table.qza" \
+  --p-sampling-depth 10000 \
```

Comment:
I wonder if we shouldn't pull this out into a variable, because it might change depending on the project.  We should also consider splitting the script into two stages, so we might inspect the read counts after demultiplexing/denoising and set this appropriately.

Reply:
I also think that we should change the --p-sampling-depth parameter should change depending on the project. This also applies to the denoising parameters.